websocket-tcp-proxy/src/main.rs

```diff
@@ -1,28 +1,71 @@
 use axum::{
     extract::{
         ws::{Message, WebSocket, WebSocketUpgrade},
-        Path, TypedHeader,
+        Path, Query, State, TypedHeader,
     },
     response::IntoResponse,
     routing::get,
     Router,
 };
-use std::env;
+use bitcoin_hashes::hex::FromHex;
+use bytes::Bytes;
+use futures::lock::Mutex;
+use serde::{de, Deserialize, Deserializer};
+use std::collections::HashMap;
 use std::net::SocketAddr;
+use std::sync::Arc;
+use std::{env, fmt, str::FromStr};
+use tokio::sync::mpsc::{self, Sender};
 use tokio::{
     io::{AsyncReadExt, AsyncWriteExt},
     net::TcpStream,
 };
 use tower_http::trace::{DefaultMakeSpan, TraceLayer};
 
+const PUBKEY_BYTES_LEN: usize = 33;
+
+pub(crate) type WSMap = Arc<Mutex<HashMap<bytes::Bytes, Sender<MutinyWSCommand>>>>;
+
+// TODO make all of these required
+#[derive(Deserialize)]
+struct MutinyConnectionParams {
+    #[serde(default, deserialize_with = "empty_string_as_none")]
+    _message: Option<String>,
+    #[serde(default, deserialize_with = "empty_string_as_none")]
+    _session_id: Option<String>,
+    #[serde(default, deserialize_with = "empty_string_as_none")]
+    _signature: Option<String>,
+}
+
+/// Serde deserialization decorator to map empty Strings to None,
+fn empty_string_as_none<'de, D, T>(de: D) -> Result<Option<T>, D::Error>
+where
+    D: Deserializer<'de>,
+    T: FromStr,
+    T::Err: fmt::Display,
+{
+    let opt = Option::<String>::deserialize(de)?;
+    match opt.as_deref() {
+        None | Some("") => Ok(None),
+        Some(s) => FromStr::from_str(s).map_err(de::Error::custom).map(Some),
+    }
+}
```

Comment:
nit: found a more generalized version online: https://github.com/serde-rs/serde/issues/1425#issuecomment-462282398

could try to do that instead incase we have more complex params in the future